### PR TITLE
Add delta endpoint

### PR DESCRIPTION
**Pull request by aliokan**
Related to : https://binnenland.atlassian.net/browse/GN-5312

- Add `/delta` endpoint
- Update README
- Split code to improve readability

BREAKING CHANGE: Update endpoints
- `/fixAnnotated` becomes `/update-all`
- `/clear-annotated` becomes `/clear`

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,409 +1,81 @@
...
-export const splitIntoChunks = (dataArray, chunkSize) => {
-  const result = [];
-  for (let i = 0; i < dataArray.length; i += chunkSize) {
-    let chunk = dataArray.slice(i, i + chunkSize);
-    result.push(chunk);
+// updates annotated templates based on insertions in the delta
+app.post("/delta", bodyParser.json({ limit: "500mb" }), async (req, res) => {
+  if (!req.body || !req.body.length) {
+    console.log("No delta found");
+    return res.status(202).send();
```
Comment (nbittich) [inline app.js]: 404 maybe?

File: app.js [hunk 2]
```diff
@@ -1,409 +1,81 @@
...
-      .map(
-        (template) => `
-      DELETE WHERE {
-        GRAPH <http://mu.semte.ch/graphs/mow/registry> { 
-          <${template.uri}> ext:annotated ?template .
-        }
-      };`
-      )
-      .join(" ")}
+  await processTemplateAnnotations(templates.results.bindings);
```
Comment (nbittich) [inline app.js]: no need to await I think, just early return to release connection with delta notifier and proceed

File: app.js [hunk 3]
```diff
@@ -1,409 +1,81 @@
...
 
-    const regex = new RegExp(`\\\${${variable.value}}`, "g");
-    const generator =
-      templateGenerators[variable.type] || templateGenerators.default;
-
-    annotatedTemplate = annotatedTemplate.replace(regex, generator(variable));
+  const updatedUris = extractInsertUris(req.body);
+  const templates = await getTemplatesAndVariables(updatedUris);
+  if (!templates.results.bindings.length) {
+    return res.send("No templates found");
```
Comment (nbittich) [inline app.js]: maybe we should just console.log it as we answer to the delta notifier and it doesn't care

**Comment by aliokan** [comment]
@nbittich I’ve incorporated your feedback. 🙏 

Looking forward to the next round! :ping_pong:

**Review by nbittich — APPROVED** [review]

**Pull request merged**